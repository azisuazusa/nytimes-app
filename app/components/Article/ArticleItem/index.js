/*
 * @flow
 */
import React, { Component } from 'react';
import { Card, CardItem, Body, Text, Grid, Col, Row } from 'native-base';
import { Image, TouchableOpacity } from 'react-native';

type Props = {
  typeOfMaterial: string,
  headline: string,
  image: string,
  bylineOriginal: string,
  pubDate: string,
  webUrl: string,
  onPress: (webUrl: string) => void
};

export default class ArticleItem extends Component<Props> {
  render() {
    const {
      typeOfMaterial,
      headline,
      image,
      bylineOriginal,
      pubDate,
      webUrl,
      onPress
    } = this.props;

    return (
      <TouchableOpacity onPress={() => onPress(webUrl)}>
        <Card>
          <CardItem header>
            <Text>{typeOfMaterial}</Text>
          </CardItem>
          <CardItem>
            <Body>
              <Grid>
                <Col><Text>{headline}</Text></Col>
                { image.length > 0 && <Col><Image source={{ uri: image }} /></Col> }
              </Grid>
            </Body>
          </CardItem>
          <CardItem footer>
            <Grid>
              <Row><Text>{bylineOriginal}</Text></Row>
              <Row><Text>{pubDate}</Text></Row>
            </Grid>
          </CardItem>
        </Card>
      </TouchableOpacity>
    );
  }
}
