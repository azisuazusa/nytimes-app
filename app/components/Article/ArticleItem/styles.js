import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  imageSize: { width: 90, height: 90 }
});
