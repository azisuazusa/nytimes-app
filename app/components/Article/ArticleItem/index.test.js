import React from 'react';
import renderer from 'react-test-renderer';
import ArticleItem from '../ArticleItem';

it('ArticleItem renders correctly', () => {
  const tree = renderer.create(
    <ArticleItem
      typeOfMaterial={'Article'}
      headline={'MAKIN DAGIN ONTA DENGAN SAYUR KOL'}
      image={'https://images.unsplash.com/photo-1547564091-6c5a23ce7aba?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80'}
      bylineOriginal={'Azis Abdul Bachar'}
      pubDate={'16 Januari 2019'}
      webUrl={'https://azisuazusa.blogspot.com'}
      onPress={() => null} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
