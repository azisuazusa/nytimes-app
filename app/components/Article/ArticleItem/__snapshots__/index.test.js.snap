// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      header={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
        uppercase={false}
      >
        Article
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            size={3}
            style={
              Object {
                "flex": 3,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              MAKIN DAGIN ONTA DENGAN SAYUR KOL
            </Text>
          </View>
          <View
            size={1}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://www.nytimes.com/https://images.unsplash.com/photo-1547564091-6c5a23ce7aba?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
                }
              }
              style={
                Object {
                  "height": 90,
                  "width": 90,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      footer={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
            uppercase={false}
          >
            Azis Abdul Bachar
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
            uppercase={false}
          >
            16 Januari 2019
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
